### ISSUE TITLE
Move source-build pipeline YAML to dotnet-release

### ISSUE THREAD

**Issue by oleksandr-didyk** [dotnet/source-build#3501]
## Motivation

We would like to move the currently public source of the source-build release pipeline into a more suitable internal repository, such as `dotnet-release`. In addition to the move itself, the pipeline would benefit from having access to Arcade infrastructure and tooling such as Secrets Manager, both of which are available and already in use in `dotnet-release`.

## Work required

We should copy the source-build pipeline YAML to the new repository + duplicate the existing AzDO pipeline until the current one can be de-commissioned. The latter task is performed with two main goals in mind:
  - the team has existing release responsibilities that could be undermined by the move
  - given the importance of the pipeline it is vital to have a reliable and tested fallback until we are certain that the move did not cause any regressions 

### Potential issues

1.   Until the new pipeline is deemed ready / functioning the team would need to support two sources, which may result in increased maintenance in addition to future changes being commited only to one of the sources
2.   New pipeline will not have all required permissions at creation. Certain access rights, such as access to certain variable groups, would need to be granted on first run. Additionally, other missing permissions may be discovered only after running the pipeline 

## Progress

- [x] existing source-build pipeline YAML has been moved (copied) to `dotnet-release`
	- https://dev.azure.com/dnceng/internal/_git/dotnet-release/pullrequest/31801 
- [x] existing AzDO pipeline has been duplicated and pointed to the copied sources
    - https://dev.azure.com/dnceng/internal/_build?definitionId=1263

Clean-up of previous pipeline is tracked via a separate issue - https://github.com/dotnet/source-build/issues/3502

**Comment by oleksandr-didyk** [comment]
Sources copied and pipeline tested with appropriate permissions to used resources granted. Links available in the description. Clean-up issue created / linked.

Closing
